todo_app/app.py

```diff
@@ -1,44 +1,83 @@
+from todo_app.user.user import Roles, User
+import flask_login
+from todo_app.github.github_authentication import GithubAuthentication
 from todo_app.mongo.mongo_wrapper import MongoWrapper
 from flask import Flask, render_template, request, redirect, url_for
 import os
 from todo_app.models.view_model import ViewModel
+from flask_login import LoginManager, login_required
 
 
 def create_app():
     app = Flask(__name__)
+    app.config["SECRET_KEY"]=os.environ["SECRET_KEY"]
     mongo_database = MongoWrapper(os.environ['DEFAULT_DATABASE'])
+    login_manager = LoginManager()
+    github_authenticator = GithubAuthentication()
 
+    @login_manager.unauthorized_handler
+    def unauthenticated():
+        authentication_url = github_authenticator.get_github_identity()
+        return redirect(authentication_url)
+        
+    
+    @login_manager.user_loader
+    def load_user(user_id):
+        if github_authenticator.authentication():
+            return User(user_id)
+        return None
+
+    login_manager.init_app(app)
 
     @app.route('/', methods=['POST'])
+    @login_required
     def add_title():
-        mongo_database.add_item(request.form.get('title'))
+        if flask_login.current_user.check_role(Roles.WRITER):
```

Comment:
We could simplify this repeated logic by pulling it out into a decorator that does this check - that would help make it simpler to update error logic if rather than redirecting we wanted to display an error page for example